src/Promotions/InPluginUpsells/SummerSalesBanner.php

```diff
@@ -0,0 +1,145 @@
+                'give'
+            ),
+            'actionText' => __('Shop Now', 'give'),
+            'alternateActionText' => __('View all StellarWP Deals', 'give'),
+            'alternateActionURL' => 'https://go.givewp.com/ss23stellar',
+            'startDate' => '2023-07-24 00:00',
+            'endDate' => '2023-07-31 23:59',
+        ];
+
+        $hasValidLicenses = self::hasValidLicenses();
+
+        if ($hasValidLicenses) {
+            return [
+                array_merge($commonBannerInfo, [
+                    'id' => 'bfgt2023givewp',
+                    'leadText' => __('Save 30% on all GiveWP Pricing Plans.', 'give'),
+                    'actionURL' => 'https://go.givewp.com/ss23give',
+                ]),
+            ];
+        }
+
+        return [
+            array_merge($commonBannerInfo, [
+                'id' => 'bfgt2023stellar',
+                'leadText' => __('Save 30% on all StellarWP products.', 'give'),
+                'actionURL' => 'https://go.givewp.com/ss23stellar',
+            ]),
+        ];
+    }
+
+
+    /**
+     * @unrleased
+     */
+    public function loadScripts()
+    {
+        wp_enqueue_script(
+            'give-in-plugin-upsells-sale-banners',
+            GIVE_PLUGIN_URL . 'assets/dist/js/admin-upsell-sale-banner.js',
+            [],
+            GIVE_VERSION,
+            true
+        );
+
+        wp_localize_script(
+            'give-in-plugin-upsells-sale-banners',
+            'GiveSaleBanners',
+            [
+                'apiRoot' => esc_url_raw(rest_url('give-api/v2/sale-banner')),
+                'apiNonce' => wp_create_nonce('wp_rest'),
+            ]
+        );
+
+        wp_enqueue_style(
+            'give-in-plugin-upsells-summer-sales-banner',
+            GIVE_PLUGIN_URL . 'assets/dist/css/admin-summer-sales-banner.css',
+            [],
+            GIVE_VERSION
+        );
+
+        wp_enqueue_style('givewp-admin-fonts');
+    }
+
+    /**
+     * @unreleased
+     */
+    public function render()
+    {
+        $banner = $this->getVisibleBanners();
+
+        if ( ! empty($banner)) {
+            include __DIR__ . '/resources/views/summer-sales-banner.php';
+        }
+    }
+
+
+    /**
+     * @unreleased
+     */
+    public static function isShowing(): bool
+    {
+        global $pagenow;
+
+        return $pagenow === 'plugins.php';
+    }
+
+    /**
+     * @unreleased
+     */
+    public static function hasValidLicenses(): bool
+    {
+        $requiredPluginSlugs = [
+            'recurring' => 'give-recurring',
+            'form_field_manager' => 'give-form-field-manager',
+            'fee_recovery' => 'give-fee-recovery',
+            'manual_donations' => 'give-manual-donations',
+            'peer_to_peer' => 'give-peer-to-peer',
+        ];
+
+        $licensedPluginSlugs = self::getLicensedPluginSlugs();
+
+
+        return sort($requiredPluginSlugs) === sort($licensedPluginSlugs);
```

Comment:
I don't have a valid license for any of these add-ons, but this function is returning true even so. 

I took a look at the PHP docs and could find it about this function: 

> Always returns true.

https://www.php.net/manual/en/function.sort.php

So, you need to use the sort function before and then return the comparison between the two arrays. To prevent any unexpected return I also would suggest replacing the `===` operator with the `==` operator that just considers the key/value pairs. More details here: https://www.php.net/manual/en/language.operators.array.php

Reply:
Hey @glaubersilva, great catch! Testing was challenging, but with Jon's help, we managed to test the retrieval of the slugs successfully. I also performed a mock test to ensure the proper rendering of different displays, and now everything is working perfectly.

```   
 Test with fake data:
 public static function hasValidLicenses(): bool
    {
        $requiredPluginSlugs = [
            'recurring' => 'give-recurring',
            'form_field_manager' => 'give-form-field-manager',
            'fee_recovery' => 'give-fee-recovery',
            'manual_donations' => 'give-manual-donations',
            'peer_to_peer' => 'give-peer-to-peer',
        ];

        $licensedPluginSlugs = [
            'give-form-field-manager',
            'give-fee-recovery',
            'give-manual-donations',
            'give-peer-to-peer',
        ];

        sort($requiredPluginSlugs);
        sort($licensedPluginSlugs);


        return $requiredPluginSlugs == $licensedPluginSlugs;
    }
```

Reply:
I swapped the ids and banner data: https://github.com/impress-org/givewp/pull/6853/commits/0484f54cff94c318de255c81cb879f5640778cee

If users have the valid licenses the stellar banner should show.